vck/src/commonMain/kotlin/at/asitplus/wallet/lib/agent/Validator.kt

```diff
@@ -1,140 +1,169 @@
 import io.matthewnelson.encoding.core.Encoder.Companion.encodeToString
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
 import kotlinx.serialization.json.buildJsonObject
-
+import kotlin.coroutines.cancellation.CancellationException
 
 /**
  * Parses and validates Verifiable Credentials and Verifiable Presentations.
  * Does verify the cryptographic authenticity of the data.
  * Does verify the revocation status of the data (when a status information is encoded in the credential).
  */
 class Validator(
-    private val verifierJwsService: VerifierJwsService = DefaultVerifierJwsService(DefaultVerifierCryptoService()),
-    private val verifierCoseService: VerifierCoseService = DefaultVerifierCoseService(DefaultVerifierCryptoService()),
+    private val verifierJwsService: VerifierJwsService = DefaultVerifierJwsService(
+        DefaultVerifierCryptoService(),
+    ),
+    private val verifierCoseService: VerifierCoseService = DefaultVerifierCoseService(
+        DefaultVerifierCryptoService(),
+    ),
     private val parser: Parser = Parser(),
-    private val zlibService: ZlibService = DefaultZlibService(),
+    private val tokenStatusResolver: (suspend (Status) -> TokenStatus)? = null,
+    private val clock: Clock = Clock.System,
 ) {
-
     constructor(
         cryptoService: VerifierCryptoService,
         parser: Parser = Parser(),
-        zlibService: ZlibService = DefaultZlibService(),
+        tokenStatusResolver: (suspend (Status) -> TokenStatus)? = null,
+        clock: Clock = Clock.System,
     ) : this(
         verifierJwsService = DefaultVerifierJwsService(cryptoService = cryptoService),
         verifierCoseService = DefaultVerifierCoseService(cryptoService = cryptoService),
         parser = parser,
-        zlibService = zlibService
+        tokenStatusResolver = tokenStatusResolver,
+        clock = clock,
     )
 
-    private var revocationList: BitSet? = null
+    constructor(
+        resolveStatusListToken: suspend (UniformResourceIdentifier) -> StatusListToken,
+        verifierJwsService: VerifierJwsService = DefaultVerifierJwsService(
+            DefaultVerifierCryptoService(),
+        ),
+        verifierCoseService: VerifierCoseService = DefaultVerifierCoseService(
+            DefaultVerifierCryptoService(),
+        ),
+        parser: Parser = Parser(),
+        zlibService: ZlibService = DefaultZlibService(),
+        clock: Clock = Clock.System,
+    ) : this(
+        verifierJwsService = verifierJwsService,
+        verifierCoseService = verifierCoseService,
+        parser = parser,
+        tokenStatusResolver = { status ->
+            val token = resolveStatusListToken(status.statusList.uri)
+
+            val payload = token.validate(
+                statusListInfo = status.statusList,
+                isInstantInThePast = {
+                    it < clock.now()
+                },
+                validateStatusListTokenIntegrity = {
+                    Validator(
```

Comment:
This seems odd, that here a new instance of `Validator` is created to perform this check

Reply:
fixed in 93ce860696aa4077acb9a053a8b3eac9bee2733b